### PR TITLE
Training the Teacher Model

### PR DESCRIPTION
**Pull request by soumik12345**
- [x] Charbonnier Loss
- [x] PSNR Metric
- [ ] SSIM Metric
- [ ] Configs as `ml_collections.ConfigDict`
- [ ] CLI for training teacher model

### CODE REVIEW

**Comment by soumik12345** [comment]
https://github.com/sayakpaul/compressed-mirnet/issues/9

File: configs/teacher.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    return config
+
+
+def get_model_configs() -> ml_collections.ConfigDict:
+    config = ml_collections.ConfigDict()
+
+    config.model_name = "teacher_model"
+    config.num_residual_recursive_groups = 3
+    config.num_multi_scale_residual_blocks = 2
+    config.num_channels = 64
```
Comment (sayakpaul) [inline configs/teacher.py:20]: If it's a teacher model, let's name the function accordingly.

File: configs/teacher.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+def get_training_configs() -> ml_collections.ConfigDict:
+    config = ml_collections.ConfigDict()
+
+    config.use_mixed_precision = False
+    config.learning_rate = 1e-4
+    config.loss_alias = "charbonnier"
+    config.charbonnier_epsilon = 1e-3
+    config.psnr_max_val = 1.0
+    config.ssim_max_val = 1.0
+    config.epochs = 100
```
Comment (sayakpaul) [inline configs/teacher.py:32]: These are the values you followed in https://github.com/soumik12345/MIRNet, right?

File: train_teacher.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+def main(_):
+    strategy = initialize_device()
+
+    # Set precision.
+    if FLAGS.experiment_configs.training_configs.use_mixed_precision:
+        mixed_precision.set_global_policy("mixed_float16")
+        logging.info("Using mixed-precision.")
+        policy = mixed_precision.global_policy()
+        assert policy.compute_dtype == "float16"
+        assert policy.variable_dtype == "float32"
```
Comment (sayakpaul) [inline train_teacher.py:40]: If this is the case, we need to ensure we're using softmax correctly. 

Please refer to this section: https://www.tensorflow.org/guide/mixed_precision#building_the_model

File: train_teacher.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+flags.DEFINE_string(
+    name="wandb_project_name", default=None, help="Name of Weights & Biases Project"
+)
+flags.DEFINE_string(
+    name="wandb_entity_name", default=None, help="Name of Weights & Biases Entity"
+)
+flags.DEFINE_string(
+    name="wandb_job_type", default=None, help="Type of Weights & Biases Job"
+)
+config_flags.DEFINE_config_file("experiment_configs")
```
Comment (sayakpaul) [inline train_teacher.py:28]: Add a usage at the beginning of this example.

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: configs/teacher.py (hunk 2 above)
Comment (soumik12345) [reply configs/teacher.py:32]: Yes
